magicctapipe/scripts/lst1_magic/semi_automatic_scripts/check_MAGIC_runs.py

```diff
@@ -0,0 +1,134 @@
+#This script allows us to get information about every MAGIC run ID (and subruns) in files (in a time interval) used for common data analysis (MAGIC1, MAGIC2, LST1). 
+
+#The MAGIC files that can be used for analysis are located in the IT cluster in the following directory:
+#/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}
+
+#In this path, 'tel_id' refers to the telescope ID, which must be either 1 or 2. 'YYYY', 'MM', and 'DD' specify the date.
+
+#In the first step, we have to load a dataframe that contains information about the date, the name of the source, and the range of MAGIC #runs. The file in file_path was generated using the spreadsheet "Common MAGIC LST1 data".
+
+import pandas as pd
+from datetime import datetime, timedelta
+import os
+import re
+
+def fix_lists_and_convert(cell):
+    # Remove brackets to avoid double lists and split on ']['
+    parts = cell.replace('][', ',').strip('[]').split(',')
+    return list(dict.fromkeys(int(item) for item in parts))
+
+def table_magic_runs(df, date_min, date_max):
+    df_selected_data = df.iloc[:, [2, 1, 25]]
+    df_selected_data.columns = ['DATE','source', 'MAGIC_runs']
+    grouped_data = df_selected_data.groupby(['DATE', 'source'])
+    result_table = []
+
+    for (date, source), group in grouped_data:
+        if (date>=date_min and date<=date_max):  
+            runs_combined = group['MAGIC_runs'].sum()
+    
+            result_table.append({
+                'DATE': date,
+                'source': source,
+                'MAGIC runs': runs_combined
+            })
+        
+    result = pd.DataFrame(result_table)
+    result['MAGIC runs'] = result['MAGIC runs'].apply(fix_lists_and_convert)
+    return(result)
+
+def existing_files( tel_id, date, source, magic_runs ):
+
+    magic_runs = str(magic_runs)  
+    date_obj = datetime.strptime(date, '%Y%m%d')
+    date_obj += timedelta(days=1)
+    new_date = datetime.strftime(date_obj, '%Y%m%d')
+    YYYY = new_date[:4]
+    MM = new_date[4:6]
+    DD = new_date[6:8]
+    Y = f"_Y_"
+    
+    path = f"/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}" 
+    
+    if os.path.exists(path):
+        files = os.listdir(path)
+        count_with_source = 0  
+        count_with_run_id = 0
+            # Counter for files that include the source.  
+            # Counter for files that include the run_id.
+        for filename in files:
+            if date and source and Y in filename:
```

Comment:
I do not think this works as you intended.
look at this example:
```
In [1]: s = "aaabbbccc"

In [6]: if "ddd" in s:
   ...:     print ("Yes")
   ...: 

In [7]: if "ddd" and "bbb" in s:
   ...:     print ("Yes")
   ...: 
Yes
```
this is because ```"ddd" and "bbb"``` returns ```"bbb"```